### DISCUSSION TITLE
Encrypting "queueName" for Azure Service Bus

### DISCUSSION THREAD

**Discussion by MoosaSaadat** [kedacore/keda#2902] [Q&A / Need Help] [answered]
Following the [example given here](https://github.com/prabdeb/sample-python-keda-service-bus-scaler/blob/main/keda-config/jobs/keda.yaml), I am setting the connection string from the secret. But, I also want to encrypt the `queueName`. How can I do that?

**Comment by tomkerkhove** [comment, marked as answer]
You cannot do that today. Curious about why you would want to do that though?

With encrypt, are you referring to the base64 hash? Because that's not encrypted.

**Reply by MoosaSaadat** [reply]
Yes, I understood later that it's not encrypted. So, now I am shifting to using Azure Key Vault. I have just started with this but I hope I can use secrets from Key Vault in keda?

**Reply by tomkerkhove** [reply]
Yes you can, but not for queue name. Out of curiosity; what is your scenario where the name of a queue is a secret?

**Reply by tomkerkhove** [reply]
Key Vault is documented on https://keda.sh/docs/2.7/concepts/authentication/, but 2.7 is still to be released

**Reply by MoosaSaadat** [reply]
I might have been mislead by the given sample: https://github.com/prabdeb/sample-python-keda-service-bus-scaler
It is creating a secret for queue name (passed to the container). So, my point was that we should either use encoded name or literal name (not a mix of both)

**Reply by MoosaSaadat** [reply]
In my case (at least) I have the same queue for Keda trigger and my container. So, following that sample, I have two different config variables to maintain although they have the same value (one encrypted, one literal). If I could use the same secret in Keda trigger, it would have reduced the redundancy and the chances of making mistakes.

**Reply by tomkerkhove** [reply]
I'm not really agreeing with that sample then and would just configure the queue name as a literal in my opinion

**Reply by MoosaSaadat** [reply]
Right. That sample is taken from the keda samples. Maybe that should be updated.
